src/coreclr/src/jit/hwintrinsic.cpp

```diff
@@ -658,9 +658,63 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
         baseType = getBaseTypeAndSizeOfSIMDType(sig->retTypeSigClass, &sizeBytes);
         retType  = getSIMDTypeForSize(sizeBytes);
         assert(sizeBytes != 0);
+
+        // Immediately return if this is non-arithmetic type.
+        if (!varTypeIsArithmetic(baseType))
+        {
+#ifdef TARGET_XARCH
+            assert(intrinsic == NI_Vector128_As || intrinsic == NI_Vector128_get_AllBitsSet ||
+                   intrinsic == NI_Vector128_get_Zero || intrinsic == NI_Vector128_WithElement ||
+                   intrinsic == NI_Vector128_ToVector256 || intrinsic == NI_Vector128_ToVector256Unsafe ||
+                   intrinsic == NI_Vector256_get_Zero || intrinsic == NI_Vector256_get_AllBitsSet ||
+                   intrinsic == NI_Vector256_As || intrinsic == NI_Vector256_WithElement ||
+                   intrinsic == NI_Vector256_GetLower);
+#else
+            assert((intrinsic == NI_Vector64_AsByte) || (intrinsic == NI_Vector128_As) ||
```

Comment:
This is smaller because the `Vector128 -> Vector64` and reverse methods aren't intrinsic yet, correct?

Reply:
Correct. As I mentioned in PR description, I started this list by running all the test cases to see for what intrinsic we would return early. All the cases for which we returned early were part of `NotSupported`. That way I made sure that we don't falsely return early for something which was supposed to be handled inside `impHWIntrinsic`.